packages/devtools_app/lib/src/service/editor/api_classes.dart

```diff
@@ -0,0 +1,432 @@
+
+  final EditorDevice device;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.device: device,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceChangedEvent &&
+      other.runtimeType == runtimeType &&
+      other.device == device;
+
+  @override
+  int get hashCode => device.hashCode;
+}
+
+/// An event sent by an editor when a device is no longer available.
+class DeviceRemovedEvent extends EditorEvent {
+  DeviceRemovedEvent({required this.deviceId});
+
+  DeviceRemovedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          deviceId: map[_Field.deviceId] as String,
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.deviceRemoved;
+
+  final String deviceId;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.deviceId: deviceId,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceRemovedEvent &&
+      other.runtimeType == runtimeType &&
+      other.deviceId == deviceId;
+
+  @override
+  int get hashCode => deviceId.hashCode;
+}
+
+/// An event sent by an editor when the current selected device was changed.
+/// This could be as a result of the client itself calling the `selectDevice`
+/// method or because the device changed for another reason (such as the user
+/// selecting a device in the editor directly, or the previously selected device
+/// is being removed).
+class DeviceSelectedEvent extends EditorEvent {
+  DeviceSelectedEvent({required this.deviceId});
+
+  DeviceSelectedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          deviceId: map[_Field.deviceId] as String?,
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.deviceSelected;
+
+  final String? deviceId;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.deviceId: deviceId,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceSelectedEvent &&
+      other.runtimeType == runtimeType &&
+      other.deviceId == deviceId;
+
+  @override
+  int get hashCode => deviceId.hashCode;
+}
+
+/// An event sent by an editor when a new debug session is started.
+class DebugSessionStartedEvent extends EditorEvent {
+  DebugSessionStartedEvent({required this.debugSession});
+
+  DebugSessionStartedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          debugSession: EditorDebugSession.fromJson(
+            map[_Field.debugSession] as Map<String, Object?>,
+          ),
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.debugSessionStarted;
+
+  final EditorDebugSession debugSession;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.debugSession: debugSession,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DebugSessionStartedEvent &&
+      other.runtimeType == runtimeType &&
+      other.debugSession == debugSession;
+
+  @override
+  int get hashCode => debugSession.hashCode;
```

Comment:
what is the reason we need to override equality and hash code for all of these? are we doing comparisons somewhere?

Reply:
Good question. We're not currently, but I started working on some integration tests for this in the DTD version (they're not in the other PR yet because I had issues making them work, because tests run in flutter_tester here and I was trying to spawn DTD).

I could remove them here and re-add them with the tests later?

Reply:
Sure, let's remove here and see if we need them later 👍

Reply:
Done! (7044c02e3)